src/JsonTxBuilderBase.sol

```diff
@@ -51,4 +52,95 @@ abstract contract JsonTxBuilderBase is CommonBase {
+    ///         This function can be overridden to provide more specific checks, but should still call
+    ///         super.checkStateDiff().
+    function checkStateDiff(Vm.AccountAccess[] memory accountAccesses) internal view virtual {
+        require(accountAccesses.length > 0, "No account accesses");
+
+        for (uint256 i; i < accountAccesses.length; i++) {
+            Vm.AccountAccess memory accountAccess = accountAccesses[i];
+            require(
+                accountAccess.account.code.length != 0,
+                string.concat("Account has no code: ", vm.toString(accountAccess.account))
+            );
+            require(
+                accountAccess.oldBalance == accountAccess.account.balance,
+                string.concat("Unexpected balance change: ", vm.toString(accountAccess.account))
+            );
+            require(
+                accountAccess.kind != VmSafe.AccountAccessKind.SelfDestruct,
+                string.concat("Self-destructed account: ", vm.toString(accountAccess.account))
+            );
+
+            for (uint256 j; j < accountAccess.storageAccesses.length; j++) {
+                Vm.StorageAccess memory storageAccess = accountAccess.storageAccesses[j];
+
+                if (!storageAccess.isWrite) continue; // Skip SLOADs.
+
+                uint256 value = uint256(storageAccess.newValue);
+                if (isLikelyAddressThatShouldHaveCode(value)) {
+                    // Log account, slot, and value if there is no code.
+                    string memory err = string.concat(
+                        "Likely address in storage has no code\n",
+                        "  account: ",
+                        vm.toString(storageAccess.account),
+                        "\n  slot:    ",
+                        vm.toString(storageAccess.slot),
+                        "\n  value:   ",
+                        vm.toString(bytes32(value))
+                    );
+                    require(address(uint160(value)).code.length != 0, err);
+                }
+
+                require(
+                    storageAccess.account.code.length != 0,
+                    string.concat("Storage account has no code: ", vm.toString(storageAccess.account))
+                );
+                require(
+                    !storageAccess.reverted,
+                    string.concat("Storage access reverted: ", vm.toString(storageAccess.account))
+                );
+                require(
+                    storageAccess.account.code.length != 0,
+                    string.concat("Storage account has no code: ", vm.toString(storageAccess.account))
+                );
+                require(
+                    !storageAccess.reverted,
+                    string.concat("Storage access reverted: ", vm.toString(storageAccess.account))
+                );
+            }
+        }
+    }
+
+    /// @notice Returns a list of addresses that are known to not have code on this chain.
```

Comment:
Nit, can we a bit more explicit on this method's purpose? Mainly to prevent "oh yea this address does have no code on mainnet, ok let me add it here so things pass" without realizing what they're doing